app/views/concerts/index.html.haml

```diff
@@ -1,22 +1,38 @@
-- title "Concert"
+- title "Concerts"
 
-%h3 Upcoming Concerts:
-%table.events-list
-  - @events.days.each do |date, events|
-    %div.event-day
-      %div.event-date
-        %h1= l date, format: :long
-        %hr
+%div.events-heading
+  %h2= link_to 'Events', events_path
+  %h2.active= link_to 'Concerts', concerts_path
+
+- @events.days.each do |date, events|
+  %div.events-container
+    %div.events-date
+      %h3= l date, format: :long
+
+
+    %div.events-day
       %ul.events
       - events.each do |event|
         %li.event
-          %h3= event.title
-          %p= l event.beginning, format: :long
+          %h4= event.title
+          -if (l event.beginning, format: '%p') != ''
+            %p= l event.beginning, format: '%l:%M %p'
+          -else
+            %p All day
```

Comment:
This is not a very clear way to check for an all day event. If it were only possible to check by comparing strings then I’d ask you to extract a helper method and give it a clear name. However, what we’re really checking here is whether this is a `TimeWithZone` or a `Date`. Switching on a class name is a code smell, though; what we really want is to be able to send the same message either way and just get what we want. See the commit I’m gonna push shortly.